modules/surface_matching/src/ppf_match_3d.cpp

```diff
@@ -539,51 +540,56 @@ void PPF3DDetector::match(const Mat& pc, std::vector<Pose3DPtr>& results, const
 
         double f[4]={0};
         computePPFFeatures(p1, n1, p2, n2, f);
-        KeyType hashValue = hashPPF(f, angle_step, distanceStep);
-
-        // we don't need to call this here, as we already estimate the tsg from scene reference point
-        // double alpha = computeAlpha(p1, n1, p2);
-        p2t[1] = tsg[1] + row2[0] * p2[0] + row2[1] * p2[1] + row2[2] * p2[2];
-        p2t[2] = tsg[2] + row3[0] * p2[0] + row3[1] * p2[1] + row3[2] * p2[2];
-
-        alpha_scene=atan2(-p2t[2], p2t[1]);
-
-        if ( alpha_scene != alpha_scene)
-        {
-          continue;
-        }
+		
+		// only search if the point is within acceptable limits
+		if (f[3]>EPS && f[3] <= this->m_diameter*1.05)
+		{
+          KeyType hashValue = hashPPF(f, angle_step, distanceStep);
+  
+          // we don't need to call this here, as we already estimate the tsg from scene reference point
+          // double alpha = computeAlpha(p1, n1, p2);
+          p2t[1] = tsg[1] + row2[0] * p2[0] + row2[1] * p2[1] + row2[2] * p2[2];
+          p2t[2] = tsg[2] + row3[0] * p2[0] + row3[1] * p2[1] + row3[2] * p2[2];
+
+          alpha_scene=atan2(-p2t[2], p2t[1]);
+
+          if ( alpha_scene != alpha_scene)
```

Comment:
Looks like impossible case. I think it make sense to remove the check(even if compiler optimize it out)

Reply:
Get the check idea. However it would be better to use `cvIsNaN(alpha_scene)` to make the code cleaner